src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -144,6 +145,10 @@ PocoHTTPClientConfiguration::PocoHTTPClientConfiguration(
 {
     /// This is used to identify configurations created by us.
     userAgent = std::string(VERSION_FULL) + VERSION_OFFICIAL;
+    if (retry_strategy.initial_delay_ms >= retry_strategy.max_delay_ms)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Initial retry delay must not exceed the maximum retry delay");
+    if (retry_strategy.jitter_factor < 0 || retry_strategy.jitter_factor > 1)
+        throw Exception(ErrorCodes::BAD_ARGUMENTS, "Jitter factor for the retry strategy must be within the [0, 1]");
```

Comment:
Would it not be better to clamp here instead of throwing? In some sense, it will silence the error... but a simple misconfiguration won't fail all the queries.